fetchcode/package_registry/__init__.py

```diff
@@ -0,0 +1,150 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from attr import attrs, attrib
+
+from packageurl.contrib.route import NoRouteAvailable
+from packageurl import PackageURL
+from packageurl.contrib.route import Router
+import requests
+
+
+router = Router()
+
+
+@attrs
+class PackageURLData:
+    """
+    Represent the Data extracted from any URL/PURL
+    `homepage_url` : URL to the homepage for this package
+    `api_url` :  Return the package repository API URL to obtain structured data for this package
+    `vcs_url` : URL for VCS download
+    `codeview_url` : URL where the code can be browsed online
+    `bugtracking_url` : URL to the issue or bug tracker for this package
+    `download_url` :  Return the package repository download URL to download the actual archive of code of this package. 
+    `documentation_url` : URL where the documentation can be found for this package
+    `readme_url` : URL where readme can be found for this package
+    `reverse_dependencies_url` : URL where reverse dependencies can be found for this package
```

Comment:
I am a little confused by this. What is an example of a `reverse_depedencies_url`?

Reply:
https://crates.io/api/v1/crates/libc/reverse_dependencies, It point's to the URL which contains, for whom package (`libc` here) is a dependency for which packages

Reply:
For example this https://crates.io/api/v1/crates/libc/reverse_dependencies